pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -438,6 +439,29 @@ public void testBloomFilterTriggering()
     }, 600_000L, "Failed to generate bloom filter");
   }
 
+  /** Check if server returns error response quickly without timing out Broker. */
+  @Test
+  public void testServerErrorWithBrokerTimeout()
```

Comment:
Can we also add a test to ensure that if there are multiple requests, the right one gets the exception on the broker side. We discussed earlier in the PR where it was difficult to identify which request threw the exception?

https://github.com/apache/incubator-pinot/pull/6596/files#r579845996

Reply:
QueryRoutingTest (unit) does some of this, but mocks the server away. I am not seeing a straightforward way of adding an integration test that would involve both server and broker since the data structures that map broker request to server response are deep down. Although, I don't think this will be a problem since all exceptions are now being handled within channelRead0 function.